staging/src/k8s.io/client-go/util/workqueue/queue.go

```diff
@@ -181,10 +185,24 @@ func (q *Type) Done(item interface{}) {
 	}
 }
 
-// ShutDown will cause q to ignore all new items added to it. As soon as the
+// Processing indicates if there are still items on the workqueue being processed.
+// It's used to drain the workqueue on an eventual shutdown
+func (q *Type) Processing() bool {
```

Comment:
I don't think this should be a public function, for any other use it basically isn't threadsafe, since a new item could get added after the unlock before the function return.

Reply:
Yes, I agree. Made it private